solution/src/main/kotlin/com/cosmotech/solution/service/SolutionServiceImpl.kt

```diff
@@ -200,6 +202,10 @@ internal class SolutionServiceImpl(
       hasChanged = true
     }
 
+    if (hasChanged && existingSolution.repository != null) {
```

Comment:
There's no need to check `hasChanged` here.
You only have to check in `solution.repository` and `solution.version` are respectively different from `existingSolution.repository` and `existingSolution.version`